lib/jsdom/living/window/Location-impl.js

```diff
@@ -11,15 +11,21 @@ const navigate = require("./navigation.js").navigate;
 
 exports.implementation = class LocationImpl {
   constructor(args, privateData) {
+    if (args.basicURL) {
+      this._basicURL = args.basicURL;
+    }
     this._relevantDocument = privateData.relevantDocument;
     this.url = null;
   }
 
   get _url() {
-    return this._relevantDocument._URL;
+    return this._basicURL || this._relevantDocument._URL;
   }
 
   _locationObjectSetterNavigate(url) {
+    if (this._basicURL) {
```

Comment:
quick way to stop writes on basic locations, since they should technically be immutable